pkg/tbtc/deposit_sweep.go

```diff
@@ -259,21 +241,12 @@ func (dsa *depositSweepAction) validateProposal(
 		// for fetching past chain events regarding the requested block
 		// range and/or returned data size. In this context, it is better to
 		// do several well-tailored calls than a single general one.
-		// We estimated the revealBlock so, we know the event was emitted
-		// at this block or somewhere close. It makes sense to establish
-		// a small margin and fetch past DepositRevealed events from range
-		// [revealBlock - margin, revealBlock + margin] in order to handle
-		// possible inaccuracies of revealBlock estimation. Moreover,
-		// we use the depositor address and wallet PKH as additional filters
-		// to limit the size of returned data.
-		revealBlockMargin := uint64(10)
-		startBlock := revealBlock - revealBlockMargin
-		endBlock := revealBlock + revealBlockMargin
-
+		// We have the revealBlock passed by the coordinator within the proposal
+		// so, we can use it to make a narrow call. Moreover, we use the
+		// wallet PKH as additional filter to limit the size of returned data.
 		events, err := dsa.chain.PastDepositRevealedEvents(&DepositRevealedEventFilter{
-			StartBlock:          startBlock,
-			EndBlock:            &endBlock,
-			Depositor:           []chain.Address{depositRequest.Depositor},
+			StartBlock:          revealBlock,
+			EndBlock:            &revealBlock,
```

Comment:
Good call on making it narrow. We could in theory allow for the `revealBlockMargin` but... why. The coordinator should pass the right data.